pkg/auth/tokens/manager.go

```diff
@@ -573,23 +573,21 @@ func (m *Manager) UserAttributeCreateOrUpdate(userID, provider string, groupPrin
 		return err
 	}
 
-	// Exists, just update if necessary
-	attribs = attribs.DeepCopy()
-
-	if m.UserAttributeChanged(attribs, provider, groupPrincipals) {
+	if needCreate {
+		attribs = attribs.DeepCopy()
```

Comment:
We need not invoke the DeepCopy() here since when needCreate=true, a new instance is just created here https://github.com/rancher/rancher/blob/647b6fa7d15cf3420a2ee82677e20fd0e4069b3e/pkg/auth/tokens/manager.go#L550

Reply:
yup, totally agree. thanks.

Reply:
fixed